[COIN-243] Get all balances for Account balance

gagbo opened the pull request.

Commit:
[cosmos] Add fees to balance history

Commit:
[cosmos] Get all cosmos assets for balance

Currently uses a lot of explorer calls, but the goal will be to use
proper data structures later nested into CosmosLikeAccount.

haammar-ledger commented on core/src/wallet/cosmos/CosmosLikeAccount.cpp:
```diff
@@ -357,14 +351,15 @@ namespace ledger {
                                                                 }
                                                                 default:
+                                                                        sum = sum - operation.fees.getValueOr(BigInt::ZERO);
                                                                         break;
```
You can just do a `sum = sum - fees` outside from the switch

gagbo replied:
Very true :D

haammar-ledger commented on core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp:
```diff
@@ -274,5 +274,153 @@ namespace ledger {
+        FuturePtr<BigInt> GaiaCosmosLikeBlockchainExplorer::getDelegatedBalance(
+            const std::string &account) const
+        {
+            const auto endpoint = fmt::format("/staking/delegators/{}/delegations", account);
```
Maybe define these URLs as constants in cosmos.hpp or somewhere ?

gagbo replied:
Will do

haammar-ledger commented on core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp:
```diff
@@ -274,5 +274,153 @@ namespace ledger {
+                            .GetArray()[0]
+                            .GetObject()["amount"]
+                            .GetString();
+                        floating_amount.erase(floating_amount.find("."), std::string::npos);
```
We may want to round instead of truncate if adding many amounts, but probably irrelevant for now.

gagbo replied:
In any case we're getting the total here, so there's nothing to sum anymore.

This result is the only one which gives non integer amounts (the amounts here are in `uatom` everytime). I think it's safe to have a balance here that's wrong by 1uatom max.

gagbo replied:
```json
{
  "height": "0",
  "result": {
    "rewards": [
      {
        "validator_address": "cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0",
        "reward": [
          {
            "denom": "uatom",
            "amount": "0.197274882002377000"
          }
        ]
      },
      {
        "validator_address": "cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn",
        "reward": [
          {
            "denom": "uatom",
            "amount": "0.334526290478806600"
          }
        ]
      }
    ],
    "total": [
      {
        "denom": "uatom",
        "amount": "0.531801172481183600"
      }
    ]
  }
}
```

haammar-ledger replied:
My bad, no addition here indeed! It's fine then.

Commit:
Move gaia endpoints to constants

Also review the rest of the review comments :
- Remove redundant substraction of fees in BalanceHistory
- Make some clang-tidy optimizations

gagbo commented:
> Looks good to me. A test for `getTotalBalance` would be ideal.

We would need a frozen account for that :( It's better to wait for live tests and then change it accordingly. Or I can add a dummy test for later
